bluecellulab/analysis/plotting.py

```diff
@@ -2,6 +2,140 @@
 
 import matplotlib.pyplot as plt
 import pathlib
+import numpy as np
+from typing import Optional, Tuple
+from bluecellulab.tools import calculate_rheobase
+from bluecellulab.circuit.circuit_access import EmodelProperties
+from bluecellulab.cell import Cell
+from bluecellulab.simulation.neuron_globals import NeuronGlobals
+from bluecellulab import Simulation
+
+
+def generate_cell_thumbnail(
+    template_path: str,
+    morphology_path: Optional[str] = None,
+    threshold_current: Optional[float] = None,
+    holding_current: Optional[float] = None,
+    template_format: str = "v6",
+    save_figure: bool = True,
+    output_path: str = "cell_thumbnail.png",
+    show_figure: bool = False,
+    v_init: Optional[float] = -80.0,
+    celsius: Optional[float] = 34.0,
+) -> Tuple[np.ndarray, np.ndarray]: 
+    """Generate a thumbnail image of a Cell object's step response.
+
+    Args:
+        template_path: Path to the hoc file.
+        morphology_path: Path to the morphology file. Default is None.
+        threshold_current: Threshold current for the cell. Default is None.
+        holding_current: Holding current for the cell. Default is None.
+        template_format: Template format. Default is "v6".
+        save_figure: Whether to save the figure. Default is True.
+        output_path: Path where to save the thumbnail image.
+        show_figure: Whether to display the figure. Default is False.
+        v_init: Initial membrane potential. Default is -80.0 mV.
+        celsius: Temperature in Celsius. Default is 34.0.
+
+    Returns:
+        Tuple containing (time, voltage) arrays of the simulation.
+    """
+    # If the threshold and holding_current is not known,
+    # we use calculate_rheobase() to find the threshold_current.
+    # For this case, we set the holding_current to 0.0 nA.
+    # if threshold_current is None: 
+    threshold_current = 1.0
```

Comment:
Why are you overwriting this value?

Reply:
The threshold current [calculation](https://github.com/openbraininstitute/BlueCelluLab/pull/19#discussion_r2161760018) should be done every time as the morphology of the me-combination will change and it won't be emodel threshold to be sure we are not using the emodel threshold. The value was overwritten just to initialise of the cell.

Reply:
this can be called after we compute the threshold, and we just get it from MEModelCalibration. So we can remove the threshold computation